upload/paymentwall.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+$startTime = microtime(true);
+$fileDir = dirname(__FILE__);
+
+require($fileDir . '/library/XenForo/Autoloader.php');
+XenForo_Autoloader::getInstance()->setupAutoloader($fileDir . '/library');
+
+XenForo_Application::initialize($fileDir . '/library', $fileDir);
+XenForo_Application::set('page_start_time', $startTime);
+
+$deps = new XenForo_Dependencies_Public();
+$deps->preLoadData();
+
+$response = new Zend_Controller_Response_Http();
+$processor = new XenForo_UserUpgradeProcessor_Paymentwall();
+$processor->initCallbackHandling(new Zend_Controller_Request_Http());
+
+$logExtra = array();
+
+try {
+    if (!$processor->validateRequest($logMessage)) {
+        $logType = 'error';
+
+        $response->setHttpResponseCode(500);
+    } else if (!$processor->validatePreConditions($logMessage)) {
+        $logType = 'error';
+    } else {
+        list($logType, $logMessage) = $processor->processTransaction();
+    }

```

Comment:
Honestly I don't think the two methods validateRequest() and validatePreConditions() are good practices. Your function should only check for errors and returns boolean, but not modify logMessage. Actually we don't need to pass log message to those methods.

the code above can be like this :
```php
try {
    if (!($processor->validateRequest() && $processor->validatePreConditions())) {
        throw new Exception($processor->getLogMessage(), 500);
    }

    list($logType, $logMessage) = $processor->processTransaction();
} catch (Exception $e) {
    $response->setHttpResponseCode($e->getCode());
    // ... 
}

$processor->log($logType, $logMessage, $logExtra);
```
in class XenForo_UserUpgradeProcessor_Paymentwall you shoud define a method to retrieve log message